app/src/main/java/dracula/vlad/androcalc/ShuntingYard.java

```diff
@@ -1,98 +1,103 @@
 package dracula.vlad.androcalc;
 
 import android.util.Log;
+
 import java.util.LinkedList;
 import java.util.Stack;
 
 class ShuntingYard {
 
-    private static String TAG = "LOG_SY";
+    private static final String TAG = ShuntingYard.class.getName();
+    private static final double ERROR_VALUE = -0.0;
+    private static final int TWO = 2;
 
     static String infixToPostfix(String infix) {
```

Comment:
to complex method, inner if/for/while operators have deep more than 3 or 4. Try to refactor and extract some inner operations to method and give clean names for them